src/panels/KaitoModelsPanel.ts

```diff
@@ -96,21 +101,103 @@ export class KaitoModelsPanelDataProvider implements PanelDataProvider<"kaitoMod
     nullIsFalse(value: boolean | null): boolean {
         return value ?? false;
     }
+    parseGPU(gpuRequirement: string): [string, number] {
+        const regex = /^Standard_NC(\d+)(ads_A100_v4|s_v3)$/;
+        const match = gpuRequirement.match(regex);
+        if (match) {
+            const cpus = parseInt(match[1], 10);
+            let family: string;
+            switch (match[2]) {
+                case "ads_A100_v4":
+                    family = "StandardNCADSA100v4Family";
+                    break;
+                case "s_v3":
+                    family = "standardNCSv3Family";
+                    break;
+                default:
+                    throw new Error("Unknown gpu format");
+            }
+
+            return [family, cpus];
+        } else {
+            throw new Error("Unknown gpu format");
+        }
+    }
 
-    private async handleDeployKaitoRequest(model: string, yaml: string, webview: MessageSink<ToWebViewMsgDef>) {
+    private async handleDeployKaitoRequest(
+        model: string,
+        yaml: string,
+        gpu: string,
+        webview: MessageSink<ToWebViewMsgDef>,
+    ) {
         this.cancelToken = false;
+        if (this.checkingGPU) {
+            return;
+        }
         try {
+            this.checkingGPU = true;
+            let quotaAvailable = false;
+            let getResult = null;
+            await longRunning(`Checking if workspace exists...`, async () => {
+                const getCommand = `get workspace workspace-${model}`;
+                getResult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, getCommand);
+            });
+            if (getResult === null || failed(getResult)) {
+                await longRunning(`Verifying available subscription quota for deployment...`, async () => {
+                    const [gpuFamily, requiredCPUs] = this.parseGPU(gpu);
+                    void requiredCPUs;
+                    const credential = new DefaultAzureCredential();
+                    const computeClient = new ComputeManagementClient(credential, this.subscriptionId);
```

Comment:
I would suggest a new utility method in file (src/commands/utils/arm.ts) to create ComputeManagementClient so that it can be used for future use cases as well. 

```
export function getComputeManagementClient(
    sessionProvider: ReadyAzureSessionProvider,
    subscriptionId: string,
): ComputeManagementClient {
    return new ComputeManagementClient(getCredential(sessionProvider), subscriptionId, { endpoint: getArmEndpoint() });
}
```